aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -771,28 +785,34 @@ _scaled_dot_product_flash_attention_cpu(
     bool is_causal,
     bool return_debug_mask,
     c10::optional<double> scale) {
-
+  const auto dtype = query.scalar_type();
   int64_t batchSize = query.size(0);
   int64_t qSize = query.size(2);
   int64_t num_head = query.size(1);
   int64_t headSize = query.size(3);
 
-  bool is_training =
-      (query.requires_grad() || key.requires_grad() ||
-      value.requires_grad());
+  TORCH_CHECK(c10::isFloatingType(dtype) && dtype != ScalarType::Half,
```

Comment:
Also, ideally these torch_checks should never get hit since we sdp_utils_cpp dispatcher knows that flash cpu implementation can not be run if any of these are set.

Reply:
Absolutely. However, it may be hit if someone directly calls `torch.ops.aten._scaled_dot_product_flash_attention`. So, I still keep the `torch_checks` here, just in case.

Reply:
Yup makes sense just making sure